apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/QueryUtils.java

```diff
@@ -0,0 +1,163 @@
+package gov.cms.bfd.server.war.stu3.providers;
+
+import ca.uhn.fhir.rest.param.DateRangeParam;
+import ca.uhn.fhir.rest.param.ParamPrefixEnum;
+import java.util.Date;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.Path;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+
+/** As set of methods to help form JPA queries. */
+public class QueryUtils {
+  /**
+   * Create a predicate for the lastUpdate field based on the passed range.
+   *
+   * @param criteriaBuilder to use
+   * @param root to use
+   * @param range to base the predicate on
+   * @return a predicate on the lastUpdated field
+   */
+  @SuppressWarnings({"rawtypes", "unchecked"})
+  static Predicate createLastUpdatedPredicate(
+      CriteriaBuilder criteriaBuilder, Root root, DateRangeParam range) {
+    final Date lowerBound = range.getLowerBoundAsInstant();
+    final Date upperBound = range.getUpperBoundAsInstant();
+    final Path lastUpdatedPath = root.get("lastUpdated");
+
+    Predicate lowerBoundPredicate = null;
+    if (lowerBound != null) {
+      switch (range.getLowerBound().getPrefix()) {
+        case GREATERTHAN:
+          lowerBoundPredicate = criteriaBuilder.greaterThan(lastUpdatedPath, lowerBound);
+          break;
+        case EQUAL:
+          lowerBoundPredicate = criteriaBuilder.greaterThanOrEqualTo(lastUpdatedPath, lowerBound);
```

Comment:
I know I'm coming in from way, way out of left field, and I don't have access to your requirements for this one, but this line is surprising in isolation (being under `case EQUAL`).  There are plenty of reasons this could be perfectly fine, but I felt like it was worth at least hanging a comment on it to call attention in case it was an accident.

Thanks for working on this.  I promise I'll be using it downstream.

Reply:
Ah, nevermind, it's the lowerbound predicate.  And you have another one on upperbound, so you're squeezing them.  Definitely fine.  Sorry to bother you.